pkg/minishift/addon/manager/addon_manager.go

```diff
@@ -358,5 +371,29 @@ func compareOpenshiftVersions(openShiftVersion, requiredOpenshiftVersion string)
 		return fmt.Errorf("\nAddon does not support OpenShift version %s. "+
 			"You need to use a version %s", openShiftVersion, requiredOpenshiftVersion)
 	}
+
+	return nil
+}
+
+func addVarDefaultsToContext(context *command.ExecutionContext, varDefaults []string) error {
+	for _, name := range varDefaults {
+		match, _ := regexp.Match(".*=.*", []byte(name))
+		if !match {
+			return errors.New(fmt.Sprintf("Var-Defaults variable must be specified in the format of <key>=<value>. '%s' is not.", name))
+		}
+
+		key, value := utilStrings.SplitKeyValue(name, "=")
```

Comment:
Why don't you use capture groups in the regexp and you get the key and value directly? I really don't get the obsession with strings.Split

Reply:
I just reused what already presenthttps://github.com/minishift/minishift/blob/40c37de0cc16651d821cccf826082fff97e6cf9e/pkg/minishift/clusterup/clusterup.go#L210-L213 .